tests/callistocore/test_views.py

```diff
@@ -787,6 +813,39 @@ def test_match_sends_custom_report(self):
         self.assertIn('test match delivery body', message.body)
         self.assertRegexpMatches(message.attachments[0][0], 'custom_.*\\.pdf\\.gpg')
 
+    @patch('callisto.delivery.views.MatchReport.encrypt_match_report')
+    def test_match_send_exception_is_handled(self, mock_encrypt_match_report):
+        mock_encrypt_match_report.side_effect = Exception('Mock Submit Match Exception')
+        response = self.client.post((self.submission_url % self.report.pk),
+                                    data={'name': 'test submitter',
+                                          'email': 'test@example.com',
+                                          'phone_number': '555-555-1212',
+                                          'email_confirmation': "False",
+                                          'key': self.report_key,
+                                          'form-0-perp': 'facebook.com/test_url',
+                                          'form-TOTAL_FORMS': '1',
+                                          'form-INITIAL_FORMS': '1',
+                                          'form-MAX_NUM_FORMS': '', })
+        self.assertIn('submit_error', response.context)
+
+    @patch('callisto.delivery.views.logger')
+    @patch('callisto.delivery.views._send_user_notification')
+    def test_match_email_confirmation_exception_is_handled(self, mock_send_user_notification, mock_logger):
+        mock_send_user_notification.side_effect = Exception('Mock Send Confirmation Exception')
+        response = self.client.post((self.submission_url % self.report.pk),
+                                    data={'name': 'test submitter',
+                                          'email': 'test@example.com',
+                                          'phone_number': '555-555-1212',
+                                          'email_confirmation': "True",
+                                          'key': self.report_key,
+                                          'form-0-perp': 'facebook.com/test_url',
+                                          'form-TOTAL_FORMS': '1',
+                                          'form-INITIAL_FORMS': '1',
+                                          'form-MAX_NUM_FORMS': '', })
+        self.assertTrue(mock_logger.exception.called)
```

Comment:
hey Michael--good call thinking about future async emails. I think mocking the call that queues an email to be sent/makes the async call would still work, bu this is a great solution too.